controllers/resource/reconciler.go

```diff
@@ -159,13 +165,16 @@ func (cor *clusterReconciler) Reconcile(ctx context.Context, objectKey types.Nam
 			resources = append(resources, r...)
 		}
 	}
-	return cor.applyTemplates(ctx, resources, dryRun)
+	return cor.applyTemplates(ctx, cs, resources, dryRun)
 }
 
-func (cor *clusterReconciler) applyTemplates(ctx context.Context, resources []*unstructured.Unstructured, dryRun bool) error {
+func (cor *clusterReconciler) applyTemplates(ctx context.Context, cs *anywherev1.Cluster, resources []*unstructured.Unstructured, dryRun bool) error {
 	for _, resource := range resources {
 		kind := resource.GetKind()
 		name := resource.GetName()
+		if cs.IsSelfManaged() && (strings.HasPrefix(name, fmt.Sprintf("%s%s", cs.Name, cpTemplatePrefix)) || strings.HasPrefix(name, fmt.Sprintf("%s%s", cs.Name, etcdTemplatePrefix))) {
+			continue
+		}
```

Comment:
wouldn't this prevent the cp template and etcd templates from ever getting updated? How do we update them? Where in the process would that happen? I'm guessing it'll happen when it's on the bootstrap cluster as part of upgrade, I'm just a bit confused where

Let's also make sure the new logic is unit tested

Reply:
After we generate the new capi spec & apply it, these templates should get updated. [Here](https://github.com/aws/eks-anywhere/blob/main/pkg/clustermanager/cluster_manager.go#L364) we are calling apply on the control plane capi content, which includes the templates for both the control plane and etcd.

& ok yes I'll be sure to add a unit test too